### PR TITLE
feat(frontend): add close button to carousel slides

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

We add a close button to the carousel slides to hide them.

# Note

This affects only current navigation. Persisting that will be applied in another PR, using the backend method `add_hidden_dapp_id` introduced with PR #3777  

# Changes

- Add button close to component `DappsCarouselSlide`.
- Bubble event `icCloseCarouselSlide` that provides the dApp that is being closed.
- Create function in `DappsCarousel` to manage the event and remove the correspondent slide.
- Adjust `Carousel` function removeSlide to refresh all needed variables.
- Add transitions.

# Tests

https://github.com/user-attachments/assets/873fa742-5307-48a2-b81a-2276e54afdc7

### CODE REVIEW

File: src/frontend/src/lib/components/carousel/Carousel.svelte [hunk 1]
```diff
@@ -250,9 +250,13 @@
...
 		slides = slides.filter((_, i) => i !== idx);
 
+		totalSlides = slides.length;
+
 		initializeCarousel();
 
-		if (slides.length === 1) {
+		goToNextSlide();
+
+		if (slides.length <= 1) {
```
Comment (DenysKarmazynDFINITY) [inline src/frontend/src/lib/components/carousel/Carousel.svelte:259]: nitpick: you can use `totalSlides` here

Comment (AntonioVentilii) [reply src/frontend/src/lib/components/carousel/Carousel.svelte]: good one! done!

File: src/frontend/src/lib/components/carousel/Carousel.svelte [hunk 2]
```diff
@@ -274,7 +279,7 @@
 	{#if nonNullish(slides) && slides.length > 1}
 		<div
 			class={`absolute bottom-2 right-0 flex justify-between px-3 ${controlsWidthStyleClass ?? 'w-full'}`}
-			transition:slide={SLIDE_PARAMS}
+			out:slide={SLIDE_PARAMS}
```
Comment (DenysKarmazynDFINITY) [inline src/frontend/src/lib/components/carousel/Carousel.svelte:282]: Wondering - wasn't `transition` covering the `out` case?

**Review by DenysKarmazynDFINITY — APPROVED** [review]
Small questions, otherwise LGTM 👍

File: src/frontend/src/lib/components/carousel/Carousel.svelte (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/carousel/Carousel.svelte:282]: yes yes, i removed the `in` transition in this way: I never noticed but it was a bit strange if you replay it slowly eheh! not noticeable too much, but you never know

**Pull request merged**